### ISSUE TITLE
Dictionary with postgres source fails

### ISSUE THREAD

**Issue by electrical** [ClickHouse/ClickHouse#23958]
Hi,

I created a dictionary with a postgres source and creation works fine. But when I try to query it, it says that the source table in postgres doesn't exist.

```
 SQL Error [156]: ClickHouse exception, code: 156, host: my.clickhouse.server, port: 8123; Code: 156, e.displayText() = DB::Exception: Failed to load dictionary 'chdatabase.dictionary_table': std::exception. Code: 1001, type: pqxx::undefined_table, e.what() = ERROR:  relation "pgschema.pgtable" does not exist
LINE 1: ...e", "iso3_code", "continent_id", "adjective" FROM "analytics...
                                                             ^
, (version 21.3.9.83 (official build))
```

The dictionary is configured using a DDL statement:

```
CREATE DICTIONARY chdatabase.dictionary_table (
	id Int32,
	name String,
	iso2_code String,
	iso3_code String
)
PRIMARY KEY id
SOURCE(POSTGRESQL(
  port 5432 
  host 'postgreshost' 
  user  'postgresuser'
  password 'postgrespassword'
  db 'pgdatabase'
  table 'pgschema.pgtable'
  )
)
LIFETIME(MIN 300 MAX 360)
LAYOUT(COMPLEX_KEY_HASHED());
```

I've confirmed that the user defined in in the postgresql source has access to the table and can query it.

**Comment by kssenii** [comment]
@electrical, it does not work because ability to specify postgres `schema` of table was added only for table engine / table function, not for dictionary source. I'll add it for dictionary also..

**Comment by electrical** [comment]
@kssenii Ahh okay :-) Thank you.
I made the wrong assumption it was in there.
Do you have an idea if this will be in an LTS release soon or should i revert back to the ODBC driver?

**Comment by kssenii** [comment]
@electrical I suppose it will not get into LTS, because it is not a bug fix..

**Comment by kssenii** [comment]
@electrical hm, I agree, may be it will get into lts.

**Comment by kssenii** [comment]
@electrical, the fix is in 21.3.

**Comment by electrical** [comment]
@kssenii thank you so much!
